### PR TITLE
Fix nil check

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
#### What type of PR is this?

/kind bug
/kind cleanup

#### What this PR does / why we need it:
While working on https://github.com/kubernetes/kubernetes/pull/101742 [noticed](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/101742/pull-kubernetes-unit/1389942934064861184) the units are failing when the surge is not configured. This PR fixes the issue by configuring the surge to 0 when maxSurge field is not configured.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/controller/daemon/util/daemonset_util.go [hunk 1]
```diff
@@ -144,6 +144,10 @@ func SurgeCount(ds *apps.DaemonSet, numberToSchedule int) (int, error) {
 	if r == nil {
 		return 0, nil
 	}
+	// If surge is not requested, we should default to 0.
+	if r.MaxSurge == nil {
```
Comment (soltysh) [inline pkg/controller/daemon/util/daemonset_util.go:148]: Can you add a unit test for this?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/controller/daemon/util/daemonset_util.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/controller/daemon/util/daemonset_util.go:148]: Sure. Modified the existing unit to run with/without featuregate flag enabled. As discussed, I'll open a follow-up PR to modify all the tests.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**